addons/sale_loyalty/models/sale_order.py

```diff
@@ -107,7 +107,7 @@ def _send_reward_coupon_mail(self):
         for order in self:
             coupons |= order._get_reward_coupons()
         if coupons:
-            coupons._send_creation_communication()
+            coupons.with_context(force_send_coupon=True)._send_creation_communication()
```

Comment:
Why a context key and not an optional parameter (defaulting to False) instead ?

```suggestion
            coupons._send_creation_communication(force_send=True)
```

Reply:
AFAIK, I thought it was not recommended to update a method signature (even for private, it should be an exception), but since you suggested, I guess it's ok :wink: Updated :+1: Thanks

Reply:
Indeed, but depends on the probability of overrides of the given method (how old the version is, ...).
I've gotten a bit sick of context keys everywhere, and try to avoid them when possible to reduce the noise.